packages/editor/src/client/panel/navigationEditPanel.tsx

```diff
@@ -0,0 +1,336 @@
+      setArticles(articleData.articles.nodes)
+    }
+  }, [articleData?.articles])
+
+  useEffect(() => {
+    const error = loadError?.message ?? createError?.message ?? updateError?.message
+    if (error) Alert.error(error, 0)
+  }, [loadError, createError, updateError])
+
+  function unionForNavigationLink(item: ListValue<NavigationLink>): NavigationLinkInput {
+    const link = item.value
+    switch (link.type) {
+      case 'Article':
+        return {
+          article: {
+            label: link.label,
+            articleID: link.articleID!
+          }
+        }
+      case 'Page':
+        return {
+          page: {
+            label: link.label,
+            pageID: link.pageID!
+          }
+        }
+      case 'External':
+        return {
+          external: {
+            label: link.label,
+            url: link.url!
+          }
+        }
+      default:
+        throw new Error('Type does not exist')
+    }
+  }
+
+  async function handleSave() {
+    if (id) {
+      const {data} = await updateNavigation({
+        variables: {
+          id,
+          input: {
+            name,
+            key,
+            links: navigationLinks.map(unionForNavigationLink)
+          }
+        }
+      })
+
+      if (data?.updateNavigation) onSave?.(data.updateNavigation)
+    } else {
+      const {data} = await createNavigation({
+        variables: {
+          input: {
+            name,
+            key,
+            links: navigationLinks.map(unionForNavigationLink)
+          }
+        }
+      })
+
+      if (data?.createNavigation) onSave?.(data.createNavigation)
+    }
+  }
+
+  return (
+    <>
+      <Drawer.Header>
+        <Drawer.Title>
+          {id ? t('navigation.panels.editNavigation') : t('navigation.panels.createNavigation')}
+        </Drawer.Title>
+      </Drawer.Header>
+      <Drawer.Body>
+        <Panel>
+          <Form fluid={true}>
+            <FormGroup>
+              <ControlLabel>{t('navigation.panels.name')}</ControlLabel>
+              <FormControl
+                placeholder={t('navigation.panels.name')}
+                value={name}
+                disabled={isDisabled}
+                onChange={value => {
+                  setName(value)
+                }}
+              />
+              <ControlLabel>{t('navigation.panels.key')}</ControlLabel>
+              <FormControl
+                placeholder={t('navigation.panels.key')}
+                value={key}
+                disabled={isDisabled}
+                onChange={value => {
+                  setKey(value)
+                }}
+              />
+            </FormGroup>
+          </Form>
+        </Panel>
+        <Panel header={t('authors.panels.links')}>
+          <ListInput
+            value={navigationLinks}
+            onChange={navigationLinkInput => setNavigationLinks(navigationLinkInput)}
+            defaultValue={{label: '', url: '', type: LinkTypes.ArticleNavigationLink}}>
```

Comment:
why does the defaultValue set the `url` property but then set the type to `ArticleNavigationLink`?

Reply:
Before setting the defaultValue for `url` I would get a warning-message that a component was changing an uncontrolled input of type text to be controlled. Setting `url` seems to me the easiest way to make the warning go away. Or do you see maybe another solution?

Reply:
I see the problem in setting the url propery and then setting the type to ArticleNavigationLink. Why don't you set a valid default.

Reply:
Yes, I did do that in the meantime. `Type` is now set as a default to `ExternalNavigationLink`